### PR TITLE
Scalaz module

### PR DESCRIPTION
**Pull request by ChernikovP**
Added `scalaz` module. Currently, it mostly mimics `cats` module but with some extra functionality.

### CODE REVIEW

File: modules/scalaz/src/main/tut/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import pureconfig.module.scalaz._
+import scalaz.{==>>, IList, ISet, Maybe, NonEmptyList}
+import scalaz.std.anyVal.intInstance
+import scalaz.std.string._
+
+case class ScalazConfig(
+  numberLst: IList[Int],
+  numberSet: ISet[Int],
+  numberNel: NonEmptyList[Int],
+  numberMap: ==>>[String, Int],
```
Comment (ruippeixotog) [inline modules/scalaz/src/main/tut/README.md]: Curiosity, isn't `String ==>> Int` the most common syntax here?

File: modules/scalaz/src/main/scala/pureconfig/module/scalaz/instances/package.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package pureconfig.module.scalaz
+
+import com.typesafe.config.ConfigValue
+import pureconfig.{ ConfigConvert, ConfigReader, ConfigWriter }
+import pureconfig.error.{ ConfigReaderFailure, ConfigReaderFailures, FailureReason }
+
+import scalaz.{ Contravariant, Equal, InvariantFunctor, MonadError, Semigroup, Show }
+
+package object instances {
```
Comment (ruippeixotog) [inline modules/scalaz/src/main/scala/pureconfig/module/scalaz/instances/package.scala:13]: Can you add some simple Scaladoc in all the objects? Copying the description from the cats module is fine.

**Review by ruippeixotog — CHANGES_REQUESTED** [review]

File: modules/scalaz/src/main/tut/README.md (hunk 1 above)
Comment (ChernikovP) [reply modules/scalaz/src/main/tut/README.md]: I think, it's the matter of being used to write `Map[A, B]`, but, anyway, `A ==>> B` looks more natural. Tried to replace it across all module in b1868cd.

File: modules/scalaz/src/main/scala/pureconfig/module/scalaz/instances/package.scala (hunk 2 above)
Comment (ChernikovP) [reply modules/scalaz/src/main/scala/pureconfig/module/scalaz/instances/package.scala:13]: Is this enough - c3543de?

File: modules/scalaz/build.sbt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+name := "pureconfig-scalaz"
+
+libraryDependencies ++= Seq(
+  "org.scalaz" %% "scalaz-core" % "7.2.27",
+  "org.scalaz" %% "scalaz-scalacheck-binding" % "7.2.27-scalacheck-1.13"
```
Comment (ruippeixotog) [inline modules/scalaz/build.sbt]: Shouldn't this be a `% "test"` dependency?

**Review by ruippeixotog — CHANGES_REQUESTED** [review]
Other than this minor thing I found now, LGTM!

File: modules/scalaz/build.sbt (hunk 3 above)
Comment (ChernikovP) [reply modules/scalaz/build.sbt]: Yep, missed that. FIxed in ef62dd9.

**Review by ruippeixotog — APPROVED** [review]

**Comment by leifwickland** [comment]
@ChernikovP, 
Thanks for your contribution. Please consider sending a PR to add yourself to the [pureconfig contributors list](https://github.com/pureconfig/pureconfig/blob/master/CONTRIBUTORS).

**Pull request merged**